value/src/test/java/com/google/auto/value/processor/ExtensionTest.java

```diff
@@ -398,6 +409,102 @@ public void testExtensionWithBuilderCompilation() throws Exception {
   }
 
   @Test
+  public void testExtensionWithBuilder() throws Exception {
+    JavaFileObject javaFileObject = JavaFileObjects.forSourceLines(
+            "foo.bar.Baz",
+            "package foo.bar;",
+            "",
+            "import com.google.auto.value.AutoValue;",
+            "import java.util.Optional;",
+            "import com.google.common.collect.ImmutableList;",
+            "",
+            "@AutoValue",
+            "public abstract class Baz {",
+            "  abstract Optional<String> foo();",
+            "  abstract String bar();",
+            "  abstract ImmutableList qux();",
+            "",
+            "  @AutoValue.Builder public static abstract class Builder {",
+            "    public abstract Builder foo(String foo);",
+            "    public abstract Builder foo(Optional<String> foo);",
+            "    public abstract Builder bar(String bar);",
+            "    public abstract ImmutableList.Builder quxBuilder();",
+            "    abstract Optional<String> foo();",
+            "    abstract Baz autoBuild();",
+            "    public Baz build() {",
+            "      if (foo().isPresent()) {",
+            "        foo(foo().get() + \" fighters\");",
```

Comment:
Not that it matters here, but could also be `ifPresent(Consumer)` instead of `isPresent`/`get`.